src/hotspot/share/services/mallocSiteTable.cpp

```diff
@@ -112,11 +115,12 @@ bool MallocSiteTable::walk(MallocSiteWalker* walker) {
  *    2. Overflow hash bucket.
  *  Under any of above circumstances, caller should handle the situation.
  */
-MallocSite* MallocSiteTable::lookup_or_add(const NativeCallStack& key, uint32_t* marker, MEMFLAGS flags) {
+MallocSite* MallocSiteTable::lookup_or_add(const NativeCallStack& key, uint16_t* bucket_idx, uint16_t* bucket_pos, MEMFLAGS flags) {
```

Comment:
Could it take a `BucketInfo*` instead of the two separate pointers?

Reply:
Maybe? I don't know how though since `BucketInfo` fields are `const` so once constructed, the fields can not change.

But I am open to suggestions here.